src/gui/systray.cpp

```diff
@@ -241,6 +241,12 @@ Systray::TaskBarPosition Systray::taskbarOrientation() const
     auto taskbarPosition = Utility::registryGetKeyValue(HKEY_CURRENT_USER,
         "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\StuckRects3",
         "Settings");
+    if (taskbarPosition.isNull()) {
```

Comment:
I think it'd be better to do a check on the key before hand using the registryKeyExists function you introduced to determine if we're going with StuckRects3 or StuckRects2. And then making a single registryGetKeyValue call with that key.

A bit like this:
```
QString taskbarPositionSubkey = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\StuckRects3";
if (!Utility::registryKeyExists(HKEY_CURRENT_USER, taskbarPositionSubkey)) {
    taskbarPositionSubkey = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Explorer\\StuckRects3";
}
auto taskbarPosition = Utility::registryGetKeyValue(HKEY_CURRENT_USER, taskbarPositionSubkey, "Settings");
```

I think it'd show the intent better.

Reply:
OK, updated it